gradoop-data-integration/src/test/java/org/gradoop/dataintegration/transformation/impl/ExtractPropertyFromVertexTest.java

```diff
@@ -229,4 +231,46 @@ public void listPropertyTest() throws Exception {
     Assert.assertTrue(properties.containsAll(Arrays.asList("m", "n", "x", "y", "z")));
   }
 
+  /**
+   * Tests graph head ids get assigned to newly created elements.
+   *
+   * @throws Exception If collect doesn't work as expected.
+   */
+  @Test
+  public void containGraphHeadIdTest() throws Exception {
+    LogicalGraph socialGraph = getSocialNetworkLoader().getLogicalGraph();
+
+    // this operation should create 6 vertices with the property values:
+    // 'Berlin', 3 times 'Dresden', 2 times 'Leipzig'
+    ExtractPropertyFromVertex extract = new ExtractPropertyFromVertex("Person", "city",
+      "City", "name", EdgeDirection.ORIGIN_TO_NEWVERTEX, "livesIn");
+
+    LogicalGraph extractedGraph = socialGraph.callForGraph(extract);
+
+    List<EPGMVertex> createdVertices = new ArrayList<>();
+    extractedGraph
+      .getVerticesByLabel("City")
+      .output(new LocalCollectionOutputFormat<>(createdVertices));
+
+    List<EPGMEdge> createdEdges = new ArrayList<>();
+    extractedGraph
+      .getEdgesByLabel("livesIn")
+      .output(new LocalCollectionOutputFormat<>(createdEdges));
+
+    // get graph heads
+    List<EPGMGraphHead> graphHeads = new ArrayList<>();
+    socialGraph.getGraphHead().output(new LocalCollectionOutputFormat<>(graphHeads));
```

Comment:
Why you are using here `socialGraph` instead of the new created `extractedGraph`?

If this is a failure, please fix.

Reply:
I wanted to check whether the newly created vertices and edges contain the graph heads of the original graph, since i thought that was the expected behavior. Do you think it is necessary to also check for the graph head in the extracted graph?

Reply:
I think this is okay, since the graph head does not change.